warp-core/src/main/scala/com/workday/warp/persistence/CoreQueries.scala

```diff
@@ -391,53 +404,127 @@ trait CoreQueries extends AbstractQueries {
 
 
   /**
-   * Creates a [[DBIO]] for inserting or updating `row` into [[TestExecutionTag]] and returning an [[Option]] with
-   * the result
-   *
+   * Creates a [[DBIO]] for inserting or updating `row` into [[TestExecutionTag]] and returning a [[DBIO]] with the
+   * row created. This updates the `value` field.
    * @param row to be inserted
-   * @return a [[DBIO]] (not yet executed) for inserting or updating `row` into [[TestExecutionTag]]
+   * @tparam T TestExecutionTagRowLikeType
+   * @return DBIO of TestExecutionTagRow updated
    */
-  override def insertOrUpdateTestExecutionTagQuery[T: TestExecutionTagRowLikeType](row: T): DBIO[Option[TestExecutionTagRowWrapper]] = {
-    TestExecutionTag returning TestExecutionTag.map(_.idTestExecutionTag) into (
-      (row, id) => row.copy(idTestExecutionTag = id)
-    ) insertOrUpdate(row)
+  override def insertOrUpdateTestExecutionTagValueQuery[T: TestExecutionTagRowLikeType](row: T): DBIO[TestExecutionTagRowWrapper] = {
+    def byId = TestExecutionTag.filter(t =>
+      t.idTestExecution === row.idTestExecution && t.idTagName === row.idTagName
+    )
+    val find: DBIO[TestExecutionTagRow] = byId.result.head
+
+    for {
+      found: Option[TestExecutionTagRow] <- byId.result.headOption
+      _ <- found match {
+        case Some(existingTag: TestExecutionTagRow) =>
+          TestExecutionTag.filter(_.idTestExecutionTag === existingTag.idTestExecutionTag)
+          .map(_.value)
+          .update(row.value)
+        case None =>
+          this.writeTestExecutionTagQuery(row)
+      }
+      tag: TestExecutionTagRow <- find
+    } yield tag
   }
 
+
   /**
    * Creates a [[DBIO]] for inserting or updating `row` into [[TestExecutionMetaTag]] and returning an [[Int]] with the
-   * rows affected
+   * rows affected. This updates the `value` field.
    *
    * @param row to be inserted
    * @return a [[DBIO]] (not yet executed) for inserting or updating `row` into [[TestExecutionMetaTag]]
    */
-  override def insertOrUpdateTestExecutionMetaTagQuery[T: TestExecutionMetaTagRowLikeType](row: T): DBIO[Int] = {
-    TestExecutionMetaTag.insertOrUpdate(row)
+  override def insertOrUpdateTestExecutionMetaTagValueQuery[T: TestExecutionMetaTagRowLikeType](row: T): DBIO[Int] = {
+    def byId(comparator: TestExecutionMetaTagRowLike) = TestExecutionMetaTag.filter(t =>
+      t.idTestExecutionTag === comparator.idTestExecutionTag && t.idTagName === comparator.idTagName
+    )
+
+    for {
+      found: Option[TestExecutionMetaTagRow] <- byId(row).result.headOption
+      metaTag: Int <- found match {
```

Comment:
lets rename this to `rowsAffected` instead of `metaTag` (also applies to `insertOrUpdateTestDefinitionMetaTagValueQuery`